test/image-blueprints/layer5-bootc/group1/rhel94-bootc-source.containerfile

```diff
@@ -1,31 +1,49 @@
 FROM registry.redhat.io/rhel9/rhel-bootc:9.4
 
 # Build arguments
-ARG PREVIOUS_MINOR_VERSION=15
 ARG USHIFT_RPM_REPO_NAME=microshift-local
 ARG USHIFT_RPM_REPO_PATH=/tmp/$USHIFT_RPM_REPO_NAME
 
 # Copy the MicroShift repository contents
 COPY $USHIFT_RPM_REPO_NAME $USHIFT_RPM_REPO_PATH
 
-# MicroShift local RPM repository
-RUN printf "\
+# Add the following repositories, using host entitlements for SSL keys:
+# - MicroShift local RPM repository
+# - Fast Datapath repository for MicroShift dependencies
+# - OpenShift previous minor version repository for MicroShift dependencies
+RUN sslkey=$(find /etc/pki/entitlement-host/ -type f -name "*-key.pem" -print -quit) && \
+    sslcrt=$(find /etc/pki/entitlement-host/ -type f -name "*.pem" ! -name "*-key.pem" -print -quit) && \
+    printf "\
 [microshift-local]\n\
 name=MicroShift Local Repository\n\
 baseurl=%s\n\
 enabled=1\n\
 gpgcheck=0\n\
-skip_if_unavailable=0\n" "${USHIFT_RPM_REPO_PATH}" > "/etc/yum.repos.d/microshift-local.repo"
-
-# OpenShift Mirror beta RPM repository for MicroShift dependencies
-# Only the released previous minor version is guaranteed to be available
-RUN printf "\
-[openshift-mirror-beta]\n\
-name=OpenShift Mirror Beta Repository\n\
-baseurl=https://mirror.openshift.com/pub/openshift-v4/%s/dependencies/rpms/4.%s-el9-beta/\n\
+skip_if_unavailable=0\n" "${USHIFT_RPM_REPO_PATH}" > "/etc/yum.repos.d/microshift-local.repo" && \
+    printf "\
+[openshift-fast-datapath]\n\
+name=Fast Datapath for RHEL 9\n\
+baseurl=https://cdn.redhat.com/content/dist/layered/rhel9/{{ .Env.UNAME_M }}/fast-datapath/os\n\
 enabled=1\n\
-gpgcheck=0\n\
-skip_if_unavailable=0\n" "$(uname -m)" "${PREVIOUS_MINOR_VERSION}" > "/etc/yum.repos.d/openshift-mirror-beta.repo"
+gpgcheck=1\n\
+gpgkey = file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release\n\
+sslverify=1\n\
+sslcacert = /etc/rhsm/ca/redhat-uep.pem\n\
+sslclientkey = %s\n\
+sslclientcert = %s\n\
+skip_if_unavailable=0\n" "${sslkey}" "${sslcrt}" > "/etc/yum.repos.d/openshift-fast-datapath.repo" && \
+    printf "\
+[openshift-rhocp]\n\
+name=OpenShift Dependencies RHEL 9\n\
+baseurl=https://cdn.redhat.com/content/dist/layered/rhel9/{{ .Env.UNAME_M }}/rhocp/4.{{ .Env.PREVIOUS_MINOR_VERSION }}/os\n\
```

Comment:
What advantages have you noticed by changing using docker's ARG into `{{ .Env.VAR }}`? Beside substitution I don't see anything that would leverage gomplate or am I mistaken?

Reply:
At the moment, we're only using substitution, but I expect this to develop further in the future as more functionality is ported to bootc. 
We cannot use ARG instead of the generic templating because we need templating in other non-container-file templates as well.